### ISSUE TITLE
Input.set_use_accumulated_input(false) broken?

### ISSUE THREAD

**Issue by TrickMe** [godotengine/godot#36664]
**Godot version:**
v3.2.stable.official

**OS/device including version:**
Linux Mint Cinnamon 19.3
also tested on: Windows 10 some new build

**Issue description:**
Input is only sent once per frame by _input(event), but i want to use the input in the moment it is triggered, without having to wait to the next frame.
So I tried "Input.set_use_accumulated_input(false)", but _input(event) is still called once per frame.

Is this a bug or is set_use_accumulated_input supposed to do something else?
There is not much in the documentation about this method.

**Steps to reproduce:**
The minimal project is set to use vsync and count the time in micro-seconds from the start of the last frame to the delivery of the input event. The measurement is displayed at the top lefthand side of the window.
The event is set to InputEventMouseButton but can also be changed.
Input.set_use_accumulated_input() is set to false.
You can set vsync off, but with vsync on it's easier to see, that the time is roughly the same with every mouse-button pressed, instead of evenly distributed from 0 to the frametime in µseconds.
If it's the same all the time, that means _input(event) is only called once per frame.
On a 60Hz display, vsync enabled, the measured time is, but shouldn't be all the time, about 16667µs.

**Minimal reproduction project:**
[inputLatency.zip](https://github.com/godotengine/godot/files/4269822/inputLatency.zip)

**Comment by SleepProgger** [comment]
Any reason this is closed @TrickMe  ?

I am running into a similar problem here.
I get multiple mouseevents per frame on linux just fine but on windows this isn't the case.
Even when setting `Engine.target_fps` to 10 i only get 1 event per frame on windows (~13 per frame on linux).

**Comment by SleepProgger** [comment]
The problem is that currently with windows the main loop sleeps. This is a problem because this leads to missed messages.
I may have a solution for it:  https://termbin.com/fybu (os_windows.cpp), but it seems pretty ugly to me.
The proper solution would be to not sleep in the message loop ofc, but that would be way more change.
If this is a proper approach (i don't think so ;)), let me know and i can send it as PR.

**Comment by ByteSizeOtter** [comment]
I am running into an issue that I feel may be affected by this. This occurs on both Windows and MacOS. I have a very simple project in which a custom `_draw()` function is set up on either a Control or Node2D with `draw_texture()` set to follow the mouse cursor. Be it this method or using a sprite or any other object set to follow the cursor, the object is always lagging 2-3 frames behind resulting in a very sluggish-feeling interface. This can be problematic for games with drag-and-drop style controls. Disabling V-sync helps some, but there is still a small lag and it's not much of a workaround. Setting `Input.set_use_accumulated_input()` to false does not appear to do anything to help this either.

**Comment by Calinou** [comment]
@ByteSizeOtter You need to use an hardware mouse cursor to avoid latency. Sending input events as soon as possible won't fix this (Godot 2.x did this, yet the 1-frame latency was still appearant).

**Comment by Calinou** [comment]
> The problem is that currently the main loop sleeps. This is a problem with windows because this leads to missed messages.
> I may have a solution for it: [termbin.com/fybu](https://termbin.com/fybu) (os_windows.cpp), but it seems pretty ugly to me.
> The proper solution would be to not sleep in the message loop ofc, but that would be way more change.
> If this is a proper approach (i don't think so ;)), let me know and i can send it as PR.

Could you repost your patch here? The link is now 404.

**Comment by SleepProgger** [comment]
@Calinou I am not working with godot since > a year or so, so no clue if it works but here you go: https://gist.github.com/SleepProgger/0e311421119294f2fb6f24ec751121c7

Its based on 3.2.1-stable.
I have no environment set up right now so it might be in a usable state, or it might not.
But even if not the idea should still work, if nothing in that area changed in godot (no clue honestly ;))

**Comment by boltut** [comment]
@Calinou I am sorry, will this bug ever fixed? I use 3.5 version of godot, and just want to draw precise line according to user input, without rendering rate limiting? What is the best way to achieve it on windows? Should I apply patch that was mentioned above or to something else? Thanks!

**Comment by akien-mga** [comment]
@boltut Did you use `Input.set_use_accumulated_input(false)`?

This issue is likely fixed in 3.5, there was some confusion in 3.4 on whether input accumulation is enabled or not, but this was fixed in 3.5 (it's enabled by default, can be disabled with that command).

**Comment by boltut** [comment]
> @boltut Did you use `Input.set_use_accumulated_input(false)`?
> 
> This issue is likely fixed in 3.5, there was some confusion in 3.4 on whether input accumulation is enabled or not, but this was fixed in 3.5 (it's enabled by default, can be disabled with that command).

@akien-mga Yes, I try to use it: "Input.set_use_accumulated_input(false)" but it is still doesn't work on windows.

**Comment by Calinou** [comment]
> and just want to draw precise line according to user input,

To draw lines that are continuous, you need to use [Bresenham's line algorithm](https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm) which can be implemented in GDScript.

**Comment by boltut** [comment]
> > and just want to draw precise line according to user input,
> 
> To draw lines that are continuous regardless of mouse speed, you need to use [Bresenham's line algorithm](https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm) which can be implemented in GDScript.

Thanks for advice, but Bresenham's algorithm doesn't solve problem in my case. I would prefer to use "Input.set_use_accumulated_input(false)" mechanism for my project. It still doesn't work in windows. My question is - can I hope to this bug will be fixed in the nearest future?

**Comment by Calinou** [comment]
> My question is - can I hope to this bug will be fixed in the nearest future?

There is no ETA for resolving issues, as contributors work on a best-effort basis. We don't even know why the issue is occurring in the first place. Also, you can disable V-Sync while the user is drawing to maximize FPS.

**Comment by boltut** [comment]
@Calinou Thanks, got it. But earlier, you had seen patch of @SleepProgger here https://github.com/godotengine/godot/issues/36664#issuecomment-845130116 May be it is possible to embed it to next maintenance release for godot 3.5.x? If it fixes problem.
